L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -40,43 +42,82 @@ class L1CTJetFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources>
   void analyze(const edm::Event&, const edm::EventSetup&) override;
   void endJob() override;
   std::vector<ap_uint<64>> encodeJets(const std::vector<l1t::PFJet> jets);
+  std::vector<ap_uint<64>> encodeSums(const std::vector<l1t::EtSum> jets);
 
-  edm::EDGetTokenT<edm::View<l1t::PFJet>> jetsToken_;
   l1t::demo::BoardDataWriter fileWriterOutputToGT_;
+  std::vector<std::pair<edm::EDGetTokenT<edm::View<l1t::PFJet>>, edm::EDGetTokenT<edm::View<l1t::EtSum>>>> tokens_;
+  std::vector<std::pair<bool, bool>> tokensToWrite_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::PFJet>>> jetsTokens_;
+  //std::vector<edm::EDGetTokenT<edm::View<l1t::EtSum>>> mhtTokens_;
 };
 
 L1CTJetFileWriter::L1CTJetFileWriter(const edm::ParameterSet& iConfig)
-    : nJets_(iConfig.getParameter<unsigned>("nJets")),
+    : collections_(iConfig.getUntrackedParameter<std::vector<edm::ParameterSet>>("collections",
```

Comment:
No need for it to be untracked, nor to specify a default as it doesn't make sense to try to run this module without any input collections.